### PR TITLE
Activity repository kj

### LINKED ISSUE #35: 3 - Dashboard - Display sleep data for recent day
For a user, their sleep data for the latest day (hours slept and quality of sleep)

### LINKED ISSUE #36: 3 - Dashboard - Display sleep date/ latest week
For a user, their sleep data over the course of the latest week (hours slept and quality of sleep)

### LINKED ISSUE #37: 3 - Dashboard - Display avg. sleep quality & hours
For a user, their all-time average sleep quality and all-time average number of hours slept

### LINKED ISSUE #79: 3 - Dashboard- Display DIY Method

### PR DESCRIPTION
**Pull request by kathrynljackson**
#### What's this PR do?
* Display data in sleep widget
* Create tests in `Activity-test.js` for the `ActivityRepository.js`

#### Where should the reviewer start?
The reviewer should start in the `index.js` file to see how the data from repositories and methods are shown on the DOM. 

#### How should this be manually tested?
These changes should manually be tested by using the calendar on the dashboard to change the date and show DOM manipulation. 

#### Any background context you want to provide?
n/a

#### What are the relevant tickets?
Closes #79 
Closes #35 
Closes #36 
Closes #37 

#### Screenshots (if appropriate)
Screen Shot 2020-09-01 at 3 24 56 PM

#### Questions:
* How can the data be shown in a way that is more user-friendly and accessible?

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -72,6 +72,10 @@ <h3 class='sleep-latest-week-data sleep-data'></h3>
            <h2 class='sleep-achievements-title sleep-title'>Achievements</h2>
            <h3 class='sleep-achievements-data sleep-data'></h3>
          </div>
+         <div class='sleep-DIY-module sleep-module'>
```
Comment (nicolegooden) [inline src/index.html:75]: We can make this more accessible by changing the `div` to `section`!

File: src/index.html [hunk 2]
```diff
@@ -98,6 +102,7 @@ <h3 class='you-vs-others-data activity-data'></h3>
     <script type='text/javascript' src='../data/activity.js'></script>
     <script type='text/javascript' src='./UserRepository.js'></script>
     <script type='text/javascript' src='./HydrationRepository.js'></script>
+    <script type='text/javascript' src='./SleepRepository.js'></script>
     <script type='text/javascript' src='./User.js'></script>
```
Comment (nicolegooden) [inline src/index.html:106]: we are also going to need to add a script element for `ActivityRepository` before we implement anything on the DOM.

File: src/scripts.js [hunk 3]
```diff
@@ -27,10 +31,13 @@ window.addEventListener('load', function actOnLoad() {
   compareStepGoals();
   getDate();
   showHydrationData();
+  showSleepData();
 });
 
 dateInput.addEventListener('change', showHydrationData);
```
Comment (nicolegooden) [inline src/scripts.js:37]: We should definitely refactor these `dateInput` event listeners the same way we did for the `window`

File: src/scripts.js [hunk 4]
```diff
@@ -84,7 +91,25 @@ function getDate() {
   return myDate;
 };
 
-function resetHydrationData() {
+function resetWidgetData() {
```
Comment (nicolegooden) [inline src/scripts.js:94]: still wondering how we can make this more dynamic.. we were going to add a parameter for which widget is being passed but got hung up on how exactly to pass in that information.  Maybe we could try passing our global query selectors?

File: src/scripts.js [hunk 5]
```diff
@@ -84,7 +91,25 @@ function getDate() {
...
   hydrationDataLatestWeek.innerText = '';
   showHydrationData();
+  sleepLatestWeekData.innerText = '';
+  showSleepData();
 }
+
+function showSleepData() {
```
Comment (nicolegooden) [inline src/scripts.js:101]: let's consider splitting this `showSleepData()` into two functions if we have time, it's super long!

File: src/index.html (hunk 1 above)
Comment (kathrynljackson) [reply src/index.html:75]: I need to do more research on what changes to make in terms of using `div`. Using more accessible layout elements is my new personal goal.

File: src/index.html (hunk 2 above)
Comment (kathrynljackson) [reply src/index.html:106]: Will the order be affected? Where should the new line go?

File: src/scripts.js (hunk 4 above)
Comment (kathrynljackson) [reply src/scripts.js:94]: I feel like we made it better with the changes, but not necessarily more dynamic. I agree that we need to keep working on it.

**Pull request merged**